content/lessons/testing/component-testing-rtl.md

```diff
@@ -0,0 +1,86 @@
+preReadQuiz:
+  description: Take this quiz to test your understanding of React Testing Library!
+  questions:
+    - description: Which of these should be used for testing asynchronous behavior?
+      choices:
+      - value: getByText()
+      - value: queryAllByRole()
+      - value: findByDisplayValue()
+      - value: queryByLabelText()
+      - value: findAllByAltText()
+      correctChoices:
+      - 2
+      - 4
+      explanation: When testing async behavior, always use the findBy* queries, which return a promise that resolves to the value of the element or elements you're querying for.
+      type: checkbox
+    - description: "True or false: If you want to inspect an element in the console, console.log is the best method."
+      choices:
+      - value: 'true'
+      - value: 'false'
+      correctChoices:
+      - 1
+      explanation: React Testing Library provides a debug() method in the object returned by render(), which will print the mounted component's DOM to the console. You can pass individual nodes to debug() to print them individually, e.g. debug(getByText('Submit'))
+      type: radio
+    - description: "Which of the following queries will return the button element in this code snippet?"
+      codeSnippet: |
+        <form>
+          <input placeholder="Name" />
+          <button type="submit">Submit</button>
+        </form>
+      choices:
+      - value: queryByText('Submit')
+      - value: queryByText(/submit/i)
+      - value: queryByText('submit')
+      - value: queryByText(/submit/)
+      - value: "queryByText('submit', {exact: false})"
+      - value: queryByRole('button')
+      correctChoices:
+      - 0
+      - 1
+      - 4
+      - 5
+      explanation: "queryByText accepts either a string or regex. If you pass a string, it needs to be exact, unless you pass {exact: false} as the second argument. If you pass regex, you need to use the `i` flag to ignore case."
+      type: checkbox
+    - description: Given the input `<input value="Hello" />` - Which of the following is the correct way to simulate a change event on the input?
+      choices:
+      - value: "queryByDisplayValue('Hello').fireEvent(change, {target: {value: 'Goodbye'}})"
+      - value: "fireEvent.change(queryByDisplayValue('Hello'), {target: {value: 'Goodbye'}})"
+      - value: "queryByText('Hello').fireEvent(change, {target: {value: 'Goodbye'}})"
+      - value: "fireEvent.onChange(queryByDisplayValue('Hello'), {target: {value: 'Goodbye'}})"
+      - value: "fireEvent.change(queryByText('Hello'), {target: {value: 'Goodbye'}})"
+      correctChoices:
+      - value: 1
+      explanation: `fireEvent` has methods for all HTML events, without the 'on' prefix. The methods each take the element you're firing the event on as the first parameter, and a mock event as an optional 2nd parameter. When selecting an input, `queryByText` will not work - use `queryByDisplayValue` or `queryByPlaceholder` instead.
```

Comment:
Gatsby doesn't like that this line starts with backticks. Could you remove them? Otherwise, the build fails and this file isn't available to view (it 404s in the list of courses.)

<img width="885" alt="Screen Shot 2019-12-09 at 1 44 09 PM" src="https://user-images.githubusercontent.com/15164473/70463130-4134a680-1a8a-11ea-93aa-57e95b8c5d9a.png">

To make sure all string formatting is included in this field, wrap the entire thing in quotation marks.


> explanation: "fireEvent has methods for all HTML events, without the 'on' prefix. The methods each take the element you're firing the event on as the first parameter, and a mock event as an optional 2nd parameter. When selecting an input, `queryByText` will not work - use `queryByDisplayValue` or `queryByPlaceholder` instead."